diff --git a/drivers/android/binder_alloc.c b/drivers/android/binder_alloc.c
index 41c0662f3..976df0167 100755
--- a/drivers/android/binder_alloc.c
+++ b/drivers/android/binder_alloc.c
@@ -214,11 +214,6 @@ static int binder_update_page_range(struct binder_alloc *alloc, int allocate,
 
 	if (mm) {
 		mmap_write_lock(mm);
-		if (!mmget_still_valid(mm)) {
-			if (allocate == 0)
-				goto free_range;
-			goto err_no_vma;
-		}
 		vma = alloc->vma;
 	}
 
diff --git a/drivers/thermal/thermal_sysfs.c b/drivers/thermal/thermal_sysfs.c
index f8bc670f1..9799afbe5 100644
--- a/drivers/thermal/thermal_sysfs.c
+++ b/drivers/thermal/thermal_sysfs.c
@@ -927,7 +927,7 @@ static void cooling_device_stats_setup(struct thermal_cooling_device *cdev)
 	var += sizeof(*stats->time_in_state) * states;
 	var += sizeof(*stats->trans_table) * states * states;
 
-	stats = kvzalloc(var, GFP_KERNEL);
+	stats = kzalloc(var, GFP_KERNEL);
 	if (!stats)
 		goto out;
 
diff --git a/mm/vmscan.c b/mm/vmscan.c
index c3af82405..6a0d62a22 100755
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -2749,6 +2749,7 @@ static void shrink_lruvec(struct lruvec *lruvec, struct scan_control *sc)
 	unsigned long nr_reclaimed = 0;
 	unsigned long nr_to_reclaim = sc->nr_to_reclaim;
 	bool proportional_reclaim;
+	bool scan_adjusted;
 	struct blk_plug plug;
 
 
